File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.js
/**
 * @param {string} s
 * @return {number}
 */
var countPalindromicSubsequence = function(s) {
    const charset = new Set();

    for(let char of s) {
        if(!charset.has(char)) {
            charset.add(char);
        }
    }

    let res = 0;

    for(let char of charset) {
        let i = 0, j = s.length - 1;

        while(s[i] != char) i++;
        while(s[j] != char) j--;
        i++;

        const charUnique = new Set();

        for(let val = i; val < j; val++) {
            charUnique.add(s[val]);
        }

        res += charUnique.size;
    }

    return res;
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        set<char> chars;

        for(auto it: s) {
            chars.insert(it);
        }

        int res = 0;

        for(auto it: chars) {
            char val = it;
            int i = 0, j = s.length() - 1;

            while(s[i] != val) i++;
            while(s[j] != val) j--;
            i++;
            set<char> unique;

            while(i < j) {
                unique.insert(s[i]);
                i++;
            }

            res += unique.size();
        }

        return res;
    }
};